rust/connlib/tunnel/src/io.rs

```diff
@@ -38,7 +41,7 @@ impl Io {
     pub fn new(
         tcp_socket_factory: Arc<dyn SocketFactory<TcpSocket>>,
         udp_socket_factory: Arc<dyn SocketFactory<UdpSocket>>,
-    ) -> io::Result<Self> {
+    ) -> Result<Self, NoInterfaces> {
```

Comment:
Do we ever match on this error?

Reply:
We don't match on it per se but it is an explicit error for passing to `on_disconnect`, that is why I wanted to make it explicit instead of just any catch-all `io::Error`.

Reply:
I think going forward, I also want to change the eventloop to fail if we can't rebind at roaming. It is unlikely to happen but I think we should be consistent in disconnecting there if we fail on startup too.